AK/Singleton.h

```diff
@@ -30,6 +31,18 @@ struct SingletonInstanceCreator {
     }
 };
 
+#ifdef KERNEL
+
+// FIXME: Find a nice way of injecting the lock rank into the singleton.
```

Comment:
Instead of templatizing this around T, should we just explicitly implement it for each instance? So the LockRank can be specified per implementation?

Reply:
I was thinking about a LockRank template argument (making it part of the type which is actually kinda nice for this use case) but my template foo is not strong enough. If you have a better solution though I'm happy to look at it, this is just a stopgap.